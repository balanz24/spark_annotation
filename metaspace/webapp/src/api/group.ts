import gql from 'graphql-tag'
import { MolecularDB } from './moldb'

export type UserGroupRole = 'INVITED' | 'PENDING' | 'MEMBER' | 'GROUP_ADMIN';
export const UserGroupRoleOptions: {[R in UserGroupRole]: R} = {
  INVITED: 'INVITED',
  PENDING: 'PENDING',
  MEMBER: 'MEMBER',
  GROUP_ADMIN: 'GROUP_ADMIN',
}
export const getRoleName = (role: UserGroupRole | null | undefined) => {
  switch (role) {
    case 'INVITED': return 'Invited'
    case 'PENDING': return 'Requesting access'
    case 'MEMBER': return 'Member'
    case 'GROUP_ADMIN': return 'Group admin'
    case null: return ''
    case undefined: return ''
  }
}

export interface CreateGroupMutation {
  createGroup: {
    id: string;
  };
}
export const createGroupMutation =
  gql`mutation createGroup($groupDetails: CreateGroupInput!) {
    createGroup(groupDetails: $groupDetails) {
      id
    }
  }`

export interface UpdateGroupMutation {
  data: {
    id: string;
    name: string;
    shortName: string;
    urlSlug: string | null;
    groupDescriptionAsHtml: string;
    currentUserRole: UserGroupRole | null;
  }
}
export const updateGroupMutation =
  gql`mutation updateGroup($groupId: ID!, $groupDetails: UpdateGroupInput!) {
    updateGroup(groupId: $groupId, groupDetails: $groupDetails) {
      id
      name
      shortName
      urlSlug
      currentUserRole
    }
  }`

export const deleteGroupMutation =
  gql`mutation deleteGroup($groupId: ID!) {
    deleteGroup(groupId: $groupId)
  }`

export const removeUserFromGroupMutation =
  gql`mutation removeUserFromGroup($groupId: ID!, $userId: ID!) {
    removeUserFromGroup(groupId: $groupId, userId: $userId)
  }`

export const requestAccessToGroupMutation =
  gql`mutation requestAccessToGroup($groupId: ID!) {
    requestAccessToGroup(groupId: $groupId) {
      role # GraphQL demands a return value be specified
    }
  }`

export const acceptRequestToJoinGroupMutation =
  gql`mutation acceptRequestToJoinGroup($groupId: ID!, $userId: ID!) {
    acceptRequestToJoinGroup(groupId: $groupId, userId: $userId) {
      role
    }
  }`

export const inviteUserToGroupMutation =
  gql`mutation inviteUserToGroup($groupId: ID!, $email: String!) {
    inviteUserToGroup(groupId: $groupId, email: $email) {
      role
    }
  }`

export const acceptGroupInvitationMutation =
  gql`mutation acceptGroupInvitation($groupId: ID!) {
    acceptGroupInvitation(groupId: $groupId) {
      role
    }
  }`

export const leaveGroupMutation =
  gql`mutation leaveGroup($groupId: ID!) {
    leaveGroup(groupId: $groupId)
  }`

export const updateUserGroupMutation =
  gql`mutation updateUserGroup($groupId: ID!, $userId: ID!, $update: UpdateUserGroupInput!) {
    updateUserGroup(groupId: $groupId, userId: $userId, update: $update)
  }`

export const importDatasetsIntoGroupMutation =
  gql`mutation($groupId: ID!, $datasetIds: [ID!]!) {
    importDatasetsIntoGroup(groupId: $groupId, datasetIds: $datasetIds)
  }`

export const editGroupQuery =
  gql`query EditGroupProfileQuery($groupId: ID!) {
    group(groupId: $groupId) {
      id
      name
      shortName
      urlSlug
      currentUserRole
      members {
        role
        numDatasets
        user {
          id
          name
          email
        }
      }
    }
  }`

export interface EditGroupQuery {
  id: string;
  name: string;
  shortName: string;
  urlSlug: string | null;
  currentUserRole: UserGroupRole | null;
  members: EditGroupQueryMember[] | null;
}
export interface EditGroupQueryMember {
  role: UserGroupRole,
  numDatasets: number,
  user: EditGroupQueryUser
}
export interface EditGroupQueryUser {
  id: string;
  name: string;
  email: string | null;
}

export interface ViewGroupResult {
  id: string;
  name: string;
  shortName: string;
  urlSlug: string | null;
  currentUserRole: UserGroupRole | null;
  numMembers: number;
  groupDescriptionAsHtml: string;
  members: ViewGroupMember[] | null;
  numDatabases: number;
}

interface ViewGroupMember {
  role: UserGroupRole;
  numDatasets: number;
  user: {
    id: string;
    name: string | null;
    email: string | null;
  }
}

export const ViewGroupFragment = gql`fragment ViewGroupFragment on Group {
  id
  name
  shortName
  urlSlug
  currentUserRole
  numMembers
  members {
    role
    numDatasets
    user { id name email }
  }
  groupDescriptionAsHtml
  numDatabases
}`

export const getGroupDatabasesQuery =
  gql`query GetGroupDatabasesQuery($groupId: ID!) {
    group(groupId: $groupId) {
      id
      numDatabases # updates tab count when database is added
      molecularDatabases {
        archived
        createdDT
        id
        isPublic
        name
        version
        user { id name email}
      }
    }
  }`

export const getUserGroupsQuery =
  gql`query GetUserGroupsQuery($query: String!, $useRole: Boolean) {
    allGroups(query: $query, limit: 1000, useRole: $useRole) {
      ...ViewGroupFragment
    }
  }${ViewGroupFragment}`

export const countGroupDatasets =
  gql`query ($groupId: ID!) { countDatasets(filter: { group: $groupId }) }`

export const countGroupsQuery =
gql`query countGroups {
  countGroups
}`

export const countPublicationsQuery =
gql`query countPublications {
  countPublications
}`

export const countReviewsQuery =
gql`query countReviews {
  countReviews
}`
