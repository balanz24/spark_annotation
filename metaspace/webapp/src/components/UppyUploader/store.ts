// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
