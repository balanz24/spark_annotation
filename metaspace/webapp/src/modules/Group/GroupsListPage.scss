$header-height: 64px;
$mini-height: 28px;

.groups-list-container{
  @apply flex flex-row flex-wrap p-0 justify-center;
  min-height: calc(100vh - #{$header-height});

  .groups-list-wrapper{
    @apply px-18 py-6 w-full max-w-4xl box-border mx-auto leading-6 text-base;

    .groups-list-header{
      @apply flex justify-between flex-wrap items-center pb-1;
      box-shadow: 0px 1px 0px 0 rgba(0, 0, 0, 0.1);

      .group-name-filter{
        //height: $mini-height;
      }

      .group-list-create-btn{
        //height: $mini-height;
      }
    }


    .groups-list-content{
      @apply flex justify-center flex-wrap items-center py-4;

      p{
        font-weight: 400;
        font-size: 14px;
        color: gray;
      }

    }


  }
}