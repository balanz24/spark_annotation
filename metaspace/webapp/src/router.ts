/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
