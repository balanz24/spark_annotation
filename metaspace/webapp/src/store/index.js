import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
