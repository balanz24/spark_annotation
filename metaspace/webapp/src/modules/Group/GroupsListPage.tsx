import { computed, defineComponent, reactive } from '@vue/composition-api'
import { Button, Input } from '../../lib/element-ui'
import { useQuery } from '@vue/apollo-composable'
import { currentUserRoleQuery, CurrentUserRoleResult } from '../../api/user'
import { getUserGroupsQuery, ViewGroupResult } from '../../api/group'
import { uniqBy } from 'lodash-es'
import GroupsListItem from './GroupsListItem'
import './GroupsListPage.scss'

interface GroupListPageProps {
  className: string
}

interface GroupListPageState {
  groupNameFilter: string | undefined
}

export default defineComponent<GroupListPageProps>({
  name: 'GroupsListPage',
  props: {
    className: {
      type: String,
      default: 'groups-list',
    },
  },
  setup: function(props, ctx) {
    const { $router } = ctx.root
    const state = reactive<GroupListPageState>({
      groupNameFilter: '',
    })

    const {
      result: currentUserResult,
    } = useQuery<CurrentUserRoleResult|any>(currentUserRoleQuery)
    const currentUser = computed(() => currentUserResult.value != null ? currentUserResult.value.currentUser
      : null)

    const queryVars = computed(() => ({
      query: state.groupNameFilter,
      useRole: true,
    }))
    const {
      result: groupsResult,
      loading: groupsLoading,
    } = useQuery<ViewGroupResult|any>(getUserGroupsQuery, queryVars)
    const groups = computed(() => groupsResult.value != null ? groupsResult.value.allGroups
      : null)

    const handleCreateGroup = () => {
      $router.push('/group/create')
    }

    return () => {
      if (!currentUser.value) {
        return (
          <div class='groups-list-container'>
            <div class='groups-list-wrapper'>
              <p class='font-normal text-center'>
                Create an account/login, to create groups and make your work reach more people!
              </p>
            </div>
          </div>)
      }

      return (
        <div class='groups-list-container'>
          <div class='groups-list-wrapper'>
            <div class='groups-list-header'>
              <Input
                class='group-name-filter tf-outer w-auto'
                value={state.groupNameFilter}
                onInput={(value: string) => {
                  state.groupNameFilter = value
                }}
                placeholder='Enter keywords'
              >
                <i
                  slot="prepend"
                  class="el-icon-search -mx-1"
                />
              </Input>
              <Button
                type="primary"
                class='group-list-create-btn'
                onClick={handleCreateGroup}
              >
                Create group
              </Button>
            </div>
            <div class='groups-list-content'>
              {
                groups.value?.length === 0
                && !groupsLoading.value
                && <p>Group not found!</p>
              }
              {
                groups.value?.length > 0
                && !groupsLoading.value
                && uniqBy((groups.value || []) as any[], 'id').map((group: any) => {
                  return <GroupsListItem
                    id={group.id}
                    name={group.name}
                    shortName={group.shortName}
                    urlSlug={group.urlSlug}
                    currentUserRole={group.currentUserRole}
                    numMembers={group.numMembers}/>
                })
              }
              {
                groupsLoading.value
                && <p><i class="el-icon-loading"/></p>
              }
            </div>
          </div>
        </div>
      )
    }
  },
})
