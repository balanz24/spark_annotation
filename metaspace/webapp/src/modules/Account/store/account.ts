import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
