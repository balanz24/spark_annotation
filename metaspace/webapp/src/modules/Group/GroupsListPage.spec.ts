import { mount } from '@vue/test-utils'
import GroupsListPage from './GroupsListPage'
import Vue from 'vue'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import store from '../../store'
import router from '../../router'
import { initMockGraphqlClient, apolloProvider } from '../../../tests/utils/mockGraphqlClient'
import { UserGroupRoleOptions } from '../../api/group'

describe('GroupsListPage', () => {
  const mockGroups = [
    {
      id: 'group1',
      name: 'group1',
      shortName: 'g1',
      numMembers: 2,
      urlSlug: 'grp1',
      currentUserRole: UserGroupRoleOptions.GROUP_ADMIN,
    },
    {
      id: 'group2',
      name: 'group2',
      shortName: 'g2',
      numMembers: 1,
      urlSlug: 'grp3',
      currentUserRole: UserGroupRoleOptions.MEMBER,
    },
  ]

  const testHarness = Vue.extend({
    components: {
      GroupsListPage,
    },
    render(h) {
      return h(GroupsListPage, { props: this.$attrs })
    },
  })

  const userQuery = () => {
    initMockGraphqlClient({
      Query: () => ({
        currentUser: () => ({ id: 'userid', role: 'user' }),
        allGroups: () => mockGroups,
      }),
    })
  }

  const userNoDataQuery = () => {
    initMockGraphqlClient({
      Query: () => ({
        currentUser: () => ({ id: 'userid', role: 'user' }),
        allGroups: () => [],
      }),
    })
  }

  const noUserQuery = () => {
    initMockGraphqlClient({
      Query: () => ({
        currentUser: () => null,
      }),
    })
  }

  beforeAll(() => {
    Vue.use(Vuex)
    sync(store, router)
  })

  beforeEach(() => {
    userQuery()
  })

  it('should match snapshot not logged', async() => {
    noUserQuery()
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot no data', async() => {
    userNoDataQuery()
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot logged', async() => {
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('should list the correct amount of groups', async() => {
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper.findAll('.group-item').length).toBe(2)
  })

  it('should change to create group page after hitting create group', async() => {
    router.replace({ path: '/groups' })
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    wrapper.find('.el-button').trigger('click')
    await Vue.nextTick()
    expect(router.currentRoute.path).toBe('/group/create')
  })

  it('should navigate to group manage page', async() => {
    router.replace({ path: '/groups' })
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    wrapper.findAll('[dataTestKey="manage-link"]').at(0).trigger('click')
    await Vue.nextTick()
    expect(router.currentRoute.name).toBe('group')
    expect(router.currentRoute.params.groupIdOrSlug).toBe(mockGroups[0].urlSlug)

    router.replace({ path: '/groups' })
    wrapper.findAll('[dataTestKey="manage-link"]').at(1).trigger('click')
    await Vue.nextTick()
    expect(router.currentRoute.name).toBe('group')
    expect(router.currentRoute.params.groupIdOrSlug).toBe(mockGroups[1].urlSlug)
  })

  it('should navigate to browse dataset page', async() => {
    router.replace({ path: '/groups' })
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    wrapper.findAll('[dataTestKey="dataset-link"]').at(0).trigger('click')
    await Vue.nextTick()
    expect(router.currentRoute.path).toBe('/datasets')
    expect(router.currentRoute.query.grp).toBe(mockGroups[0].id)

    router.replace({ path: '/groups' })
    wrapper.findAll('[dataTestKey="dataset-link"]').at(1).trigger('click')
    await Vue.nextTick()
    expect(router.currentRoute.path).toBe('/datasets')
    expect(router.currentRoute.query.grp).toBe(mockGroups[1].id)
  })
})
