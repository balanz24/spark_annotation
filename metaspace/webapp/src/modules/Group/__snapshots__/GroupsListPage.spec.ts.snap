// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsListPage should match snapshot logged 1`] = `
<div class="groups-list-container">
  <div class="groups-list-wrapper">
    <div class="groups-list-header">
      <div class="el-input el-input-group el-input-group--prepend group-name-filter tf-outer w-auto">
        <div class="el-input-group__prepend"><i class="el-icon-search -mx-1"></i></div><input type="text" autocomplete="off" placeholder="Enter keywords" class="el-input__inner">
        <!---->
        <!---->
        <!---->
        <!---->
      </div><button type="button" class="el-button el-button--primary group-list-create-btn">
        <!---->
        <!----><span>Create group</span></button>
    </div>
    <div class="groups-list-content">
      <div class="group-item">
        <div class="group-item-info">
          <div class="group-item-title-wrapper group-item-info-line"><a href="/group/grp1" class="group-item-title" datatestkey="group-link">group1<span class="group-item-short-name"> (g1)</span></a>
            <mock-el-popover trigger="hover" placement="top" popper-class="min-w-0" class="ml-1">
              <div slot-key="default">
                <p class="leading-none m-0">Copy group id to clipboard</p>
              </div>
              <div slot-key="reference"><button class="button-reset flex  w-6 h-6"><svg data-icon="copy-id" class="h-full w-full text-gray-400 hover:text-gray-500 sm-copy-icon"></svg></button></div>
            </mock-el-popover>
          </div>
          <div class="group-item-info-line"><a href="/datasets?grp=group1" class="group-item-title">66 Datasets,</a>2 Members</div>
        </div>
        <div class="group-item-actions">
          <div><i class="el-icon-picture"></i><a href="/datasets?grp=group1" class="ml-1" datatestkey="dataset-link">Browse datasets</a></div>
          <div><i class="el-icon-edit"></i><a href="/group/grp1?tab=settings" class="ml-1" datatestkey="manage-link">Manage group</a></div>
        </div>
      </div>
      <div class="group-item">
        <div class="group-item-info">
          <div class="group-item-title-wrapper group-item-info-line"><a href="/group/grp3" class="group-item-title" datatestkey="group-link">group2<span class="group-item-short-name"> (g2)</span></a>
            <mock-el-popover trigger="hover" placement="top" popper-class="min-w-0" class="ml-1">
              <div slot-key="default">
                <p class="leading-none m-0">Copy group id to clipboard</p>
              </div>
              <div slot-key="reference"><button class="button-reset flex  w-6 h-6"><svg data-icon="copy-id" class="h-full w-full text-gray-400 hover:text-gray-500 sm-copy-icon"></svg></button></div>
            </mock-el-popover>
          </div>
          <div class="group-item-info-line"><a href="/datasets?grp=group2" class="group-item-title">66 Datasets,</a>1 Member</div>
        </div>
        <div class="group-item-actions">
          <div><i class="el-icon-picture"></i><a href="/datasets?grp=group2" class="ml-1" datatestkey="dataset-link">Browse datasets</a></div>
          <div><i class="el-icon-edit"></i><a href="/group/grp3?tab=members" class="ml-1" datatestkey="manage-link">Browse group</a></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GroupsListPage should match snapshot no data 1`] = `
<div class="groups-list-container">
  <div class="groups-list-wrapper">
    <div class="groups-list-header">
      <div class="el-input el-input-group el-input-group--prepend group-name-filter tf-outer w-auto">
        <div class="el-input-group__prepend"><i class="el-icon-search -mx-1"></i></div><input type="text" autocomplete="off" placeholder="Enter keywords" class="el-input__inner">
        <!---->
        <!---->
        <!---->
        <!---->
      </div><button type="button" class="el-button el-button--primary group-list-create-btn">
        <!---->
        <!----><span>Create group</span></button>
    </div>
    <div class="groups-list-content">
      <p>Group not found!</p>
    </div>
  </div>
</div>
`;

exports[`GroupsListPage should match snapshot not logged 1`] = `
<div class="groups-list-container">
  <div class="groups-list-wrapper">
    <p class="font-normal text-center">Create an account/login, to create groups and make your work reach more people!</p>
  </div>
</div>
`;
